src/lib/model/OutboundTransfersModel.js

```diff
@@ -520,6 +524,103 @@ class OutboundTransfersModel {
         });
     }
 
+    /**
+     * Get transfer details by sending GET /transfers request to the switch;
+     * then waits for a notification from the cache that the transfer has been resolved.
+     */
+    async _getTransfer() {
```

Comment:
@kirgene I think you have misunderstood this story a bit. GET /transfer call from DFSP backend to switch should pass through the scheme-adapter, sure you need to turn async to sync but the transfer state must not come from the cache! The single source of truth for the transfer is the switch. This method must ALWAYS call the switch; your lines 537-548 are superfluous; there may NEVER be state in the cache for the requested transfer, that is a normal use case, scheme-adapter should ALWAYS forward the request to the switch even if it has never seen that ID before.